File: app/src/main/java/com/ed/redditapp/ui/post_detail/PostDetailPresenter.java
package com.ed.redditapp.ui.post_detail;

public class PostDetailPresenter {
    private final PostDetailActivity view;
    private final PostDetailModel model;

    public PostDetailPresenter(PostDetailActivity view, PostDetailModel model) {
        this.view = view;
        this.model = model;
    }

    public void onActivityLoaded(String postUrl) {
        model.getPostComments(postUrl, view::updateComments);
    }
}

File: app/src/main/java/com/ed/redditapp/ui/post_detail/PostDetailPresenter.kt
package com.ed.redditapp.ui.post_detail

class PostDetailPresenter(
        private val view: PostDetailActivity,
        private val model: PostDetailModel) {
    fun onActivityLoaded(postUrl: String) {
        model.getPostComments(postUrl, view::updateComments)
    }
}